### PR TITLE
Add basic docs sidebar

### PR DESCRIPTION
**Pull request by kevinkassimo**
Add basic closable sidebar for navigation.

deno_web_1

Also introduced `node-sass` to make writing styles easier.

### CODE REVIEW

**Comment by kevinkassimo** [comment]
Hmm it seems we are modifying on the same files that would eventually result in a bunch of merge conflicts in the end...

File: src/Docs.tsx [hunk 1]
```diff
@@ -22,43 +24,56 @@ interface Props {
...
+      <nav className="sidebar">
+        <Drawer
+          variant="persistent"
+          anchor={'left'}
+          open={drawerOpen}
+          onClose={setDrawerOpen.bind(null, false)}
+        >
+          <div className="header">
+            <Button onClick={setDrawerOpen.bind(null, false)}>Close</Button>
+            <h2>Menu</h2>
```
Comment (ry) [inline src/Docs.tsx]: Superfluous

File: src/Docs.tsx [hunk 2]
```diff
@@ -22,43 +24,56 @@ interface Props {
...
-      <nav>
-        <List>
-          {docs.map(d => {
+    <div className="docs">
+      <nav className="sidebar">
+        <Drawer
+          variant="persistent"
+          anchor={'left'}
+          open={drawerOpen}
+          onClose={setDrawerOpen.bind(null, false)}
```
Comment (ry) [inline src/Docs.tsx]: It doesn't need to open and close

File: src/Docs.tsx [hunk 3]
```diff
@@ -22,43 +24,56 @@ interface Props {
...
-                <Markdown source={d.docstr} />
-                <Divider />
-                <CodeBlock language="json" value={JSON.stringify(d, null, 1)} />
-              </CardContent>
-            </Card>
-          </Box>
-        );
-      })}
+      <main>
+        {<Button onClick={() => setDrawerOpen(!drawerOpen)}>{drawerOpen ? "Hide Menu" : "Show Menu"}</Button>}
```
Comment (ry) [inline src/Docs.tsx]: Remove

File: src/Docs.scss [hunk 4]
```diff
@@ -0,0 +1,13 @@
+.docs {
+  .sidebar {
+    min-width: 100px;
+    max-width: 200px;
+    @media screen and (max-width: 800px) {
+      display: none;
+    }
+    .header {
+      margin: 10px auto 0 auto;
+      text-align: center;
```
Comment (ry) [inline src/Docs.scss]: Are the .header ones necessary? I'm trying to minimize subjective changes to css.

File: src/Docs.scss [hunk 5]
```diff
@@ -0,0 +1,13 @@
+.docs {
+  .sidebar {
+    min-width: 100px;
+    max-width: 200px;
+    @media screen and (max-width: 800px) {
+      display: none;
```
Comment (ry) [inline src/Docs.scss]: It would be good if the TOC was just at the top of the mobile webpage, rather than having a close/open action.

**Comment by kevinkassimo** [comment]
Closing this. There are a lot of subjective design decisions that could be only materialized after an initial complete design is done. IMHO PRs of this kind turns out to be less productive. I'll instead discuss about design suggestions in the chatroom instead.

**Pull request closed without merging**